src/main/java/com/project/service/CommentServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+
+    @Override
+    public List<CommentResponse> getAll(Long appId) {
+        List<Comment> comments = new ArrayList<>(commentRepository.findAll());
+        return comments.stream().filter(comment -> comment.getApplication().getId().equals(appId)).map(CommentResponse::new)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public CommentResponse getById(Long id) throws CommentNotFoundException {
+        Comment comment = commentRepository.findById(id).orElseThrow(() -> new CommentNotFoundException(id));
+        return new CommentResponse(comment);
+    }
+
+    @Override
+    public byte[] getAttachment(Long id, String filename) throws CommentNotFoundException,
+            CommentAttachmentNotFoundException {
+        Comment comment = commentRepository.findById(id).orElseThrow(() -> new CommentNotFoundException(id));
+
+        if (!comment.getAttachmentName().equals(filename)) {
+            throw new CommentAttachmentNotFoundException(id);
+        }
+
+        return comment.getAttachment();
+    }
+
+    @Override
+    public void addAttachment(Long id, MultipartFile file) throws CommentNotFoundException, IOException, InvalidFileException {
+        Comment comment = commentRepository.findById(id).orElseThrow(() -> new CommentNotFoundException(id));
+        byte[] bytes = file.getBytes();
+        String extension = FilenameUtils.getExtension(file.getOriginalFilename());
+
+        if (file.getContentType() != null && !file.getContentType().trim().equals("")){
+            if(!allowedContentTypes.contains(file.getContentType()))
+                throw new InvalidFileException(file.getContentType());
+        }
+
+        if (!allowedFileExtensions.contains(extension))
+            throw new InvalidFileException(extension);
+
+        comment.setAttachment(bytes);
+        comment.setAttachmentName(file.getOriginalFilename());
+        commentRepository.save(comment);
+    }
+
+    @Override
+    public List<CommentResponse> getApplicantVisibleComments(Long appId) {
+        List<CommentResponse> comments = getAll(appId);
+        return comments.stream().filter(CommentResponse::isVisibleToApplicant).collect(Collectors.toList());
+    }
+
+    @Override
+    public CommentResponse addAdminComment(CommentRequest comment, Long appId)
+            throws ApplicationNotFoundException, UserNotFoundException {
+        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        boolean isExternal = comment.isVisibleToApplicant();
+        Application app = applicationService.getById(appId);
+        User author = userService.getByEmail(comment.getAuthorEmail());
+
+        if (isExternal) {
+            app.setNewExternalComment(true);
+            app.setLastExternalCommentAuthor(comment.getAuthorEmail());
+            if (author.getRoles().contains(UserRole.ADMIN.name())) {
+                emailService.sendEmail(
+                        app.getApplicant().getEmail(), "Gavote naują komentarą",
```

Comment:
This can be moved to a properties file since this is just a plain text.

Reply:
stored plain text in `messages.properties`